add rest api Gen2 (#7225)
aws-amplify/docs: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/test-api/index.mdx

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
                  
export const meta = {
  title: 'Test the REST API',
  description: 'Learn how you can test the REST API from the terminal, with Amplify Mock, or with the API Gateway console.',
  platforms: [
    'android',
    'angular',
    'flutter',
    'javascript',
    'nextjs',
    'react',
    'react-native',
    'swift',
    'vue'
  ],
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta
    }
  };
}
          

## Test the API from the terminal

> If Guest users have access to your REST API you can test it from the terminal using Curl.

Reviewer: Curl -> curl or cURL
https://curl.se/

> Curl is available in many distributions including Mac, Windows and Linux. Follow the install instructions in the [docs](https://curl.haxx.se/docs/install.html).

<BlockSwitcher>
<Block name="Mac and Linux">

### GET method example

```bash
curl https://a5b4c3d2e1.execute-api.eu-west-2.amazonaws.com/dev/items
```

### POST method example

```bash
curl -H "Content-Type: application/json" -d '{"name":"item-1"}' https://a5b4c3d2e1.execute-api.eu-west-2.amazonaws.com/dev/items
```

</Block>
<Block name="Windows">

### GET method example

```bash
curl https://a5b4c3d2e1.execute-api.eu-west-2.amazonaws.com/dev/items
```

### POST method example

```bash
curl -H "Content-Type: application/json" -d {\"name\":\"item-1\"} https://a5b4c3d2e1.execute-api.eu-west-2.amazonaws.com/dev/items
```

</Block>
</BlockSwitcher>

> Important! Testing methods using production endpoints may result in changes to resources that cannot be undone.

## Test the API with API Gateway console

Let's test your new REST API using the route below with HTTP Method `GET` and path `/items?limit=10` which includes a `limit` query string parameter.

```console
GET /items?limit=10
```

> > Important! Testing methods with the API Gateway console may result in changes to resources that cannot be undone.

Reviewer: can we expand on this and use a callout instead of a blockquote?

> - Sign in to the API Gateway console at [https://console.aws.amazon.com/apigateway](https://console.aws.amazon.com/apigateway).

Reviewer: ```suggestion
- Sign in to the [API Gateway console](https://console.aws.amazon.com/apigateway)
```

> - Choose the `myRestApi` REST API.
> - In the Resources pane, choose the method you want to test. Pick `ANY` right under `/items`.

Reviewer: ```suggestion
- Choose the `myRestApi` REST API
- In the Resources pane, choose the method you want to test. Select `ANY` right under `/items`
```

bullet points should not end with periods unless they are in sentence form
"pick" -> "select" (or "choose")

```console
/                        
 |_ /items               Main resource. Eg: /items
   GET                   Methods
   DELETE
   PUT
   POST
   OPTIONS               Allow pre-flight requests in CORS by browser
    |_ /{proxy+}         Proxy resource. Eg: /items/, /items/id, items/object/{id}
    ANY                  Includes methods: DELETE, GET, HEAD, OPTIONS, PATCH, POST, PUT
    OPTIONS              Allow pre-flight requests in CORS by browser
```

> - In the Method Execution pane, in the Client box, choose TEST. Choose the `GET` method. Add `limit=10` to the Query String `{items}` field.

Reviewer: ```suggestion
- In the Method Execution pane, in the Client box, choose TEST. Choose the `GET` method. Add `limit=10` to the query string `{items}` field.
```

what is the "client box"?

> 
> - Choose Test to run the test for `GET /items?limit=10`. The following information will be displayed: request, status, latency, response body, response headers and logs.

Reviewer: ```suggestion
- Choose Test to run the test for `GET /items?limit=10`. The following information will be displayed: request, status, latency, response body, response headers, and logs.
```

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
                  
export const meta = {
  title: 'Test the REST API',
  description: 'Learn how you can test the REST API from the terminal, with Amplify Mock, or with the API Gateway console.',
  platforms: [
    'android',
    'angular',
    'flutter',
    'javascript',
    'nextjs',
    'react',
    'react-native',
    'swift',
    'vue'
  ],
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta
    }
  };
}
          

## Test the API from the terminal

> If Guest users have access to your REST API you can test it from the terminal using curl.
> 
> [Curl](https://github.com/curl/curl) is a command-line tool that lets you transfer data to and from a server using various protocols.

Reviewer: ```suggestion
If unauthenticated guest users have access to your REST API you can test it from the terminal using [curl]((https://github.com/curl/curl)). curl is a command-line tool that lets you transfer data to and from a server using various protocols.
```

> > Curl is available in many distributions including Mac, Windows and Linux. Follow the install instructions in the [docs](https://curl.haxx.se/docs/install.html).

Reviewer: ```suggestion
> curl is available in many distributions including Mac, Windows, and Linux. [Follow the install instructions in the docs](https://curl.haxx.se/docs/install.html).
```

<BlockSwitcher>
<Block name="Mac and Linux">

### GET method example

```bash title="Terminal" showLineNumbers={false} 
curl https://a5b4c3d2e1.execute-api.eu-west-2.amazonaws.com/dev/items
```

### POST method example

```bash title="Terminal" showLineNumbers={false} 
curl -H "Content-Type: application/json" -d '{"name":"item-1"}' https://a5b4c3d2e1.execute-api.eu-west-2.amazonaws.com/dev/items
```

</Block>
<Block name="Windows">

### GET method example

```bash title="Terminal" showLineNumbers={false} 
curl https://a5b4c3d2e1.execute-api.eu-west-2.amazonaws.com/dev/items
```

### POST method example

```bash title="Terminal" showLineNumbers={false} 
curl -H "Content-Type: application/json" -d {\"name\":\"item-1\"} https://a5b4c3d2e1.execute-api.eu-west-2.amazonaws.com/dev/items
```

</Block>
</BlockSwitcher>

<Callout warning="true">

> **Warning:** Testing methods using production endpoints may result in changes to resources that cannot be undone.

Reviewer: what kind of changes result from testing the methods? or is this referring to unintentionally setting data in your prod env?

Author: the warning is geared to towards the data modification the call might make

</Callout>

## Test the API with API Gateway console

Let's test your new REST API using the route below with HTTP Method `GET` and path `/items?limit=10` which includes a `limit` query string parameter.

```console
GET /items?limit=10
```

<Callout warning="true">

**Warning:** Testing methods with the API Gateway console may result in changes to resources that cannot be undone.

</Callout>

> - Sign in to the [API Gateway console](https://console.aws.amazon.com/apigateway).
> - Choose the `myRestApi` REST API.
> - In the Resources pane, choose the method you want to test. Select `GET` right under `/items`.
> 
> ```console
> /                        
>  |_ /items               Main resource. Eg: /items
>    GET                   Methods
>    DELETE
>    PUT
>    POST
>    OPTIONS               Allow pre-flight requests in CORS by browser
>     |_ /{proxy+}         Proxy resource. Eg: /items/, /items/id, items/object/{id}
>     ANY                  Includes methods: DELETE, GET, HEAD, OPTIONS, PATCH, POST, PUT
>     OPTIONS              Allow pre-flight requests in CORS by browser
> ```
> 
> - In the Method Execution pane, select **TEST**. Choose the `GET` method and add `limit=10` to the query string `{items}` field.
> 
> - Choose **Test** to run the test for `GET /items?limit=10`. The following information will be displayed: request, status, latency, response body, response headers and logs.

Reviewer: ```suggestion
1. Sign in to the [API Gateway console](https://console.aws.amazon.com/apigateway)
2. Choose the `myRestApi` REST API
3. In the Resources pane, choose the method you want to test. Select `GET` right under `/items`.

  ```console
  /                        
  |_ /items               Main resource. Eg: /items
    GET                   Methods
    DELETE
    PUT
    POST
    OPTIONS               Allow pre-flight requests in CORS by browser
      |_ /{proxy+}         Proxy resource. Eg: /items/, /items/id, items/object/{id}
      ANY                  Includes methods: DELETE, GET, HEAD, OPTIONS, PATCH, POST, PUT
      OPTIONS              Allow pre-flight requests in CORS by browser
  ```

4. In the Method Execution pane, select **TEST**. Choose the `GET` method and add `limit=10` to the query string `{items}` field.
5. Choose **Test** to run the test for `GET /items?limit=10`. The following information will be displayed: request, status, latency, response body, response headers and logs.
```

since we're listing steps this should be represented as an ordered list
